addons/base/models.py

```diff
@@ -319,7 +318,7 @@ def merge(self, user_settings):
         except KeyError:
             pass
         else:
-            connected = Model.find(Q('user_settings', 'eq', user_settings))
+            connected = Model.objects.filter(user_settings=user_settings)
```

Comment:
Similar to other note, I think you can use `update()` here and remove the loop below.